### PR TITLE
Static analysis fixes and dependency version bumps

### PR DESCRIPTION
**Pull request by chas678**
* Automated Fixes found by running FindBugs and IntelliJ analysis tools
* Upgrade dependency versions found by `mvn versions:display-dependency-updates`
* Unit tests for exception flows changed to use @Rule ExpectedException so can test message
* Remove deprecated JSR310 Jackson module in Jackson replaced JavaTime module

/cc @mxns + @MrShish

### CODE REVIEW

File: izettle-cart/src/main/java/com/izettle/cart/ItemLine.java [hunk 1]
```diff
@@ -86,9 +87,7 @@ public String toString() {
             + " item = " + item
             + ", grossValue = " + grossValue
             + ", grossVat = " + grossVat
-            + ", actualValue = " + actualValue
-            + ", actualVat = " + actualVat
-            + '}';
+            + ", actualValue = " + actualValue + ", actualVat = " + actualVat + '}';
```
Comment (ssprang) [inline izettle-cart/src/main/java/com/izettle/cart/ItemLine.java]: I liked the other format better, feel like changing it back?

File: izettle-cassandra/src/main/java/com/izettle/astyanax/impl/AwsSecurityGroupHostSupplier.java [hunk 2]
```diff
@@ -91,10 +93,13 @@ private Region findMyRegion() {
             conn.setConnectTimeout(DEFAULT_TIMEOUT); // 5 seconds
             conn.setReadTimeout(DEFAULT_TIMEOUT); // 5 seconds
 
-            try (BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream()))) {
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(
+                conn.getInputStream(),
+                Charset.forName("UTF-8")
```
Comment (ssprang) [inline izettle-cassandra/src/main/java/com/izettle/astyanax/impl/AwsSecurityGroupHostSupplier.java]: Could we use java.nio.charset.StandardCharsets.UTF_8 here instead?

File: izettle-cart/src/main/java/com/izettle/cart/CartUtils.java [hunk 3]
```diff
@@ -117,7 +117,7 @@ static Double getDiscountPercentage(long grossValue, Long discountValue) {
         if (discountValue == null) {
             return null;
         } else {
-            return 100d * discountValue / grossValue;
+            return 100.0d * discountValue / grossValue;
```
Comment (MrShish) [inline izettle-cart/src/main/java/com/izettle/cart/CartUtils.java]: is this really needed when using doubles?

Comment (chas678) [inline izettle-cart/src/main/java/com/izettle/cart/CartUtils.java]: compiler doesn’t complain - but adding the decimal makes it less confusing
- and it shuts up the analyse warning

On 29 February 2016 at 13:59, Umut Yurdakök <notifications@github.com>
wrote:

> In izettle-cart/src/main/java/com/izettle/cart/CartUtils.java
> <https://github.com/iZettle/izettle-toolbox/pull/163#discussion_r54405958>
> :
>
> > @@ -117,7 +117,7 @@ static Double getDiscountPercentage(long grossValue, Long discountValue) {
> >          if (discountValue == null) {
> >              return null;
> >          } else {
> > -            return 100d * discountValue / grossValue;
> > +            return 100.0d * discountValue / grossValue;
>
> is this really needed when using doubles?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/iZettle/izettle-toolbox/pull/163/files#r54405958>.
>



-- 
––
*Christopher Barham*
Developer
iZettle

+46 73-531 82 29
www.izettle.com <http://izettle.com/>

**Comment by mxns** [comment]
Nice work @chas678

**Pull request merged**